test/CMakeLists.txt

```diff
@@ -7,20 +7,26 @@ if(MSVC)
     set(CMAKE_STATIC_LIBRARY_PREFIX "")
 endif()
 
+if (CMAKE_GENERATOR STREQUAL "Ninja")
+	set (CMAKE_DEBUG_POSTFIX "d")
+endif ()
+
 ExternalProject_Add(
 	googletest_project
-	SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/gtest-1.8.0"
+	SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/gtest-1.8.1"
 	INSTALL_DIR "${CMAKE_CURRENT_BINARY_DIR}/prefix"
 	CMAKE_ARGS
 		-DCMAKE_INSTALL_PREFIX:PATH=<INSTALL_DIR>
 		-DBUILD_GMOCK=ON
 		-Dgtest_force_shared_crt=ON
+	BUILD_BYPRODUCTS
+		"<INSTALL_DIR>/lib/${CMAKE_STATIC_LIBRARY_PREFIX}gmock${CMAKE_DEBUG_POSTFIX}${CMAKE_STATIC_LIBRARY_SUFFIX}"
 )
 
 add_library(gmock UNKNOWN IMPORTED)
 set_target_properties(gmock PROPERTIES
     IMPORTED_LOCATION
-    ${PROJECT_BINARY_DIR}/test/prefix/lib/${CMAKE_STATIC_LIBRARY_PREFIX}gmock${CMAKE_STATIC_LIBRARY_SUFFIX}
+		${PROJECT_BINARY_DIR}/test/prefix/lib/${CMAKE_STATIC_LIBRARY_PREFIX}gmock${CMAKE_DEBUG_POSTFIX}${CMAKE_STATIC_LIBRARY_SUFFIX}
```

Comment:
This seems surprising. Why does it need this postfix? Why debug, given that this line isn't specific to debug?

Reply:
Gtest sets the postfix.

Reply:
For my projects I use automatic download of gtest and compilation at a configuration time but I am not sure if you'd like that.
Also the configuration and the build fail in VS2019 without the above changes.

Reply:
(I prefer to vendor the dep here.)

So gtest sets the postfix, but only for debug and ninja builds?